shared/src/price_estimate.rs

```diff
@@ -34,7 +34,11 @@ pub enum PriceEstimationError {
 
 #[async_trait::async_trait]
 pub trait PriceEstimating: Send + Sync {
-    // Price is given in how much of sell_token needs to be sold for one buy_token.
+    /// For OrderKind::Sell amount is in sell_token and for OrderKind::Buy in buy_token.
+    /// The resulting price is how many units of sell_token needs to be sold for one unit of
+    /// buy_token (sell_amount / buy_amount).
+    /// If amount is 0 then kind is ignored and the result is how many units of buy token you get
```

Comment:
Oh wow, this is pretty bad (and totally my fault when implementing this). Agree that we should probably not handle 0 amounts in the same method or at least make it that they return it in the same unit.